packages/astro/src/core/render-context.ts

```diff
@@ -566,3 +573,31 @@ export class RenderContext {
 		});
 	}
 }
+
+/**
+ * Return the component path without the `srcDir` and `pages`
+ * @param component
+ */
+function getAstroRoutePattern(component: RouteData['component']): string {
+	let splitComponent = component.split("/");
+	while (true) {
+		const currentPart = splitComponent.shift();
+		if (!currentPart) {break}
+		
+		// "pages" isn't configurable, so it's safe to stop here
+		if (currentPart === "pages") {
```

Comment:
What if the component from an integration contains "pages"?

Reply:
It would work, because we exist the first time we see `pages`. So `src/pages/blog/pages/last.astro` will be `blog/pages/last`